akka-persistence-query/src/main/scala/akka/persistence/query/Hint.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.persistence.query
+
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * A query hint that defines how to execute the query,
+ * typically specific to the journal implementation.
+ */
+trait Hint
+
+/**
+ * If the underlying datastore only supports queries that are completed when they reach the
+ * end of the "result set", the journal has to submit new queries after a while in order
+ * to support "infinite" event streams that include events stored after the initial query has completed.
+ *
+ * A plugin may optionally support this [[Hint]] for defining such a refresh interval.
```

Comment:
“A plugin may optionally support this hint” is probably true for all hints; it should be enough to document this on the `trait Hint` itself.